### PR TITLE
validate,wasm: remove use of slow binary.Read

### PR DESCRIPTION
**Pull request by sbinet**
This CL removes the calls to binary.Read(r, bo, &v) where v is a simple
numeric type.
It replaces it with explicit calls to binary.Order.Uint{32|64} that
don't involve calls to reflect.

---
This change is [Reviewable](https://reviewable.io/reviews/go-interpreter/wagon/17)

### CODE REVIEW

**Comment by vibhavp** [comment]
:lgtm:

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/go-interpreter/wagon/17#-:-Kr15-pRa-rhvKuOMdnE:bnfp4nl)*

File: wasm/read.go [hunk 1]
```diff
@@ -25,3 +26,21 @@ func readString(r io.Reader, n int) (string, error) {
...
+	var buf [4]byte
+	_, err := io.ReadFull(r, buf[:])
+	if err != nil {
+		return 0, err
+	}
+	return binary.LittleEndian.Uint32(buf[:]), nil
+}
+
+func readU64(r io.Reader) (uint64, error) {
+	var buf [4]byte
```
Comment (shurcooL) [inline wasm/read.go]: Shouldn't this be `var buf [8]byte`? `uint64` is 8 bytes.

Comment (sbinet) [inline wasm/read.go]: yes. good catch. copy-paste hit me yet again :)

Comment (shurcooL) [inline wasm/read.go]: Followup question, how come tests didn't catch this? I'd expect things to fail badly as a result of a low level primitive like this being incorrect. Is this code not used/tested significantly yet? (That's completely fine if so, I just want to learn.)

Comment (sbinet) [inline wasm/read.go]: addressed w/ go-interpreter/wagon#18

Comment (sbinet) [inline wasm/read.go]: b/c there isn't any *yet*. at least not for `wasm` itself.
(I was in the process of adding some. see go-interpreter/wagon#19)

Comment (shurcooL) [inline wasm/read.go]: Makes sense. Thanks!

**Pull request merged**